plugins/modules/nmcli.py

```diff
@@ -2934,31 +2953,71 @@ def main():
 
         elif nmcli.state == "up":
             if nmcli.connection_exists():
-                if module.check_mode:
-                    module.exit_json(changed=True)
-                if nmcli.conn_reload:
-                    (rc, out, err) = nmcli.reload_connection()
-                (rc, out, err) = nmcli.up_connection()
-                if rc != 0:
-                    module.fail_json(name=f"Error bringing up connection named {nmcli.conn_name}", msg=err, rc=rc)
+                is_active = nmcli.is_connection_active()
+
+                if is_active and not nmcli.conn_reload:
+                    result["changed"] = False
+                    result["msg"] = f"Connection {nmcli.conn_name} is already active"
+                    module.exit_json(**result)
+                else:
+                    if module.check_mode:
+                        module.exit_json(changed=True, **result)
+
+                    if nmcli.conn_reload:
+                        (rc, out, err) = nmcli.reload_connection()
+                        if rc != 0:
+                            module.fail_json(
+                                name=f"Error bringing up connection named {nmcli.conn_name}", msg=err, rc=rc
+                            )
+
+                    if not is_active or nmcli.conn_reload:
```

Comment:
This `if` is inside the `else` (2962) for the `if is_active and not nmcli.conn_reload` (2958).

Meaning that this `if` (2973) will ***always*** be `True`, therefore, it is redundant.